internal/util/go_with_wait_group_test.go

```diff
@@ -0,0 +1,40 @@
+package util_test
+
+import (
+	"github.com/redhat-et/apex/internal/util"
+	"github.com/stretchr/testify/assert"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+func TestGoWithWaitGroup(t *testing.T) {
+
+	// Verify you can use a nil WaitGroup
+	util.GoWithWaitGroup(nil, func() {
+	})
+
+	// Verify all goroutines are done before the WaitGroup counter is zero.
+	wg := &sync.WaitGroup{}
+	for i := 0; i < 10; i++ {
+		// run a few functions in parallel
+		util.GoWithWaitGroup(wg, func() {
+			time.Sleep(200 * time.Millisecond)
+		})
+	}
+
+	done := atomic.Bool{}
+	go func() {
+		wg.Wait()
+		done.Store(true)
+	}()
+
+	// should not be done yet
+	assert.False(t, done.Load())
```

Comment:
It seems *possible* for this to fail (unlikely, I know), and the assert.Eventually() you have below still checks to make sure that the waitgroup successfully finished. I'd just drop this assert.